docs/nightly/en/user-guide/continuous-aggregation/manage-flow.md

```diff
@@ -0,0 +1,59 @@
+# Manage Flows
+
+Each `flow` is a continuous aggregation query in GreptimeDB. It is a query that continuously updates the aggregated data based on the incoming data and materializes the result. This document describes how to create, update, and delete a flow.
+
+A `flow` have those attributes:
+- `name`: the name of the flow. It's an unique identifier in the catalog level.
+- `source tables`: tables provide data for the flow. Each flow can have multiple source tables.
+- `sink table`: the table to store the materialized aggregated data.
+- `expire after`: the interval to expire the data from the source table. Data after the expiration time will not be used in the flow.
+- `comment`: the description of the flow.
+- `SQL`: the continuous aggregation query to define the flow. Refer to [Expression](./expression.md) for the available expressions.
+
+# Create or update a flow
+
+The grammar to create a flow is:
+
+```sql
+CREATE [ OR REPLACE ] FLOW [ IF NOT EXISTS ] <name>
+OUTPUT TO <sink-table-name>
+[ EXPIRE AFTER <expr> ]
+[ COMMENT = "<string>" ]
+AS 
+<SQL>;
+```
+
+When `OR REPLACE` is specified, if a flow with the same name already exists, it will be updated to the new one.
+
+`sink-table-name` is the table name to store the materialized aggregated data. It can be an existing table or a new table, `flow` will create the sink table if it doesn't exist. But if the table already exists, the schema of the table must match the schema of the query result.
+
+`expire after` is an optional interval to expire the data from the source table. The expiration time is a relative time from the current time (by "current time" we means the physical time of the data arrive the Flow engine). For example, `INTERVAL '1 hour'` means the data after 1 hour from the current time will be expired. Expired will be dropped directly.
```

Comment:
> For example, `INTERVAL '1 hour'` means the data after 1 hour from the current time will be expired. Expired will be dropped directly.

This is strange. At current time, how to expire the data in the future?

Reply:
> > For example, `INTERVAL '1 hour'` means the data after 1 hour from the current time will be expired. Expired will be dropped directly.
> 
> This is strange. At current time, how to expire the data in the future?

There is currently no way to expire data in the future, as the usecase is expected to be very much contemporary, but this is prone to change in the future

Reply:
@discord9 cannot understand what you said. This comment just wants to know how the `expire` works.

Reply:
> @discord9 cannot understand what you said. This comment just wants to know how the `expire` works.

So currently this expire args only matter to reduce operator with time index(i.e. group by tumble), it will treat said column as event timestamp, and expire rows with event timstamp older than now(system time) for that given expire duration, admittedly this is only a temporal solution to streaming process's state management, but for now its just that, and works with expected usecase(without too many late data arrival that is).

Reply:
I change the wording to "older" (versus "newer"/"future"). "Before" and "after" are misleading.